### PR TITLE
Service Registration, Heartbeat, and Discovery

### PR DESCRIPTION
**Pull request by jakepeyser**
This feature branch incorporates service discovery into versions of the controller running on PROD. All other running instances will use env vars as their target service endpoints.

### CODE REVIEW

**Comment by cphrmky** [comment]
> 1. I created service-discovery.py to handle the SD operations. I didn't create any classes (like a ServicePublisher) as I thought it would be excessive. Do you think the situation warrants that sort of design?
 
I'm thinking maybe yes because of #2.
 
> 2. ... This works well, but since there is no object created I lose control over that thread and have no way of gracefully ending the while loop that emits the heartbeats.  ...
 
I like this solution.  http://stackoverflow.com/questions/323972/is-there-any-way-to-kill-a-thread-in-python
 
You can hook your de-register work into the stop() method and you should be good to go. So that you're calling thusly somewhere in your graceful exit:

```python
t.stop()
t.join()
```
 
Overall this code looks :+1: to me.

File: .env.local [hunk 1]
```diff
@@ -1,4 +1,7 @@
+export ERP_SERVICE=http://0.0.0.0:3000
 export SMTP_USER_NAME=XXX
 export SMTP_PASSWORD=XXX
 export SMTP_SERVER=smtp.gmail.com
-export SMTP_SERVER_PORT=587
\ No newline at end of file
+export SMTP_SERVER_PORT=587
+export SD_URL=https://servicediscovery.ng.bluemix.net
+export SD_AUTH=XXX
```
Comment (cphrmky) [inline .env.local]: git is complaining about a missing `\n` at the EOF

Comment (jakepeyser) [inline .env.local]: @cphrmky is it best practice to have a `\n` at the end of non-`.py` files as well?

**Comment by jakepeyser** [comment]
Makes sense, thanks for the pointers @cphrmky. I am going to try to decouple the code more, as well, in the hopes of publishing it as a client on PyPi

File: .env.local (hunk 1 above)
Comment (jakepeyser) [inline .env.local]: [This SO question/answer](http://stackoverflow.com/questions/5813311/no-newline-at-end-of-file#answer-5813359) sums it up for me, newline it is

Comment (cphrmky) [inline .env.local]: yup

**Pull request merged**